simulator-starter/src/main/java/com/consol/citrus/simulator/http/HttpScenarioActionBuilder.java

```diff
@@ -12,37 +16,58 @@
 
     /** Scenario endpoint */
     private final ScenarioEndpoint scenarioEndpoint;
-
-    /** Spring application context */
-    private ApplicationContext applicationContext;
-
+    
     public HttpScenarioActionBuilder(ScenarioEndpoint scenarioEndpoint) {
         this.scenarioEndpoint = scenarioEndpoint;
     }
 
+    /**
+     * Sets the bean reference resolver.
+     * @param referenceResolver
+     */
+    public HttpScenarioActionBuilder withReferenceResolver(ReferenceResolver referenceResolver) {
+        super.withReferenceResolver(referenceResolver);
+        return this;
+    }
+    
     /**
      * Default scenario receive operation.
      * @return
      */
-    public HttpServerActionBuilder.HttpServerReceiveActionBuilder receive() {
-        return new HttpServerActionBuilder(action, scenarioEndpoint)
-                .withApplicationContext(applicationContext)
-                .receive();
+    public HttpServerReceiveActionBuilder receive() {
+    	HttpServerReceiveActionBuilder receiveActionBuilder = server(scenarioEndpoint).receive();
+    	return receiveActionBuilder;
     }
 
     /**
      * Default scenario send response operation.
      * @return
      */
-    public HttpServerActionBuilder.HttpServerSendActionBuilder send() {
-        return new HttpServerActionBuilder(action, scenarioEndpoint)
-                .withApplicationContext(applicationContext)
-                .send();
+    public HttpServerSendActionBuilder send() {
+    	HttpServerSendActionBuilder sendActionBuilder = server(scenarioEndpoint).send();
+    	return sendActionBuilder;
     }
 
-    @Override
-    public HttpScenarioActionBuilder withApplicationContext(ApplicationContext applicationContext) {
-        this.applicationContext = applicationContext;
-        return (HttpScenarioActionBuilder) super.withApplicationContext(applicationContext);
+    
+    public HttpServerActionBuilder server(ScenarioEndpoint scenarioEndpoint) {
+    	
+    	// TODO: change HttpActionBuilder to accept endpoint in calls to server
+    	// Endpoint cannot be used to create a server anymore although underlying builders 
+    	// would accept it. Citrus 3.0.0 needs to be adjusted to make this possible again. 
+    	// For now use an ugly reflection hack to inject the endpoint.
```

Comment:
I assume this is a thing on our roadmap (by that I mean PF internal)?

Reply:
This has to be solved in citrus. It is one of the open issues mentioned in the header comment.